modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/IgniteAbstractTxSuspendResumeTest.java

```diff
@@ -0,0 +1,771 @@
+
+        cfgs.add(cacheConfiguration("cache1", PARTITIONED, 0, false));
+        cfgs.add(cacheConfiguration("cache2", PARTITIONED, 1, false));
+        cfgs.add(cacheConfiguration("cache3", PARTITIONED, 1, true));
+        cfgs.add(cacheConfiguration("cache4", REPLICATED, 0, false));
+
+        if (!FORCE_MVCC)
+            cfgs.add(cacheConfiguration("cache5", LOCAL, 0, false));
+
+        return cfgs;
+    }
+
+    /**
+     * @param cacheMode Cache mode.
+     * @param backups Number of backups.
+     * @param nearCache If {@code true} near cache is enabled.
+     * @return Cache configuration.
+     */
+    private CacheConfiguration<Integer, Integer> cacheConfiguration(
+        String name,
+        CacheMode cacheMode,
+        int backups,
+        boolean nearCache) {
+        CacheConfiguration<Integer, Integer> ccfg = new CacheConfiguration<>(name);
+
+        ccfg.setCacheMode(cacheMode);
+        ccfg.setAtomicityMode(TRANSACTIONAL);
+        ccfg.setWriteSynchronizationMode(FULL_SYNC);
+
+        if (cacheMode == PARTITIONED)
+            ccfg.setBackups(backups);
+
+        if (nearCache)
+            ccfg.setNearConfiguration(new NearCacheConfiguration<Integer, Integer>());
+
+        return ccfg;
+    }
+
+    /**
+     * @param c Closure.
+     */
+    protected void executeTestForAllCaches(CI2<Ignite, IgniteCache<Integer, Integer>> c) {
+        for (int i = 0; i < gridCount(); i++) {
+            Ignite ignite = ignite(i);
+
+            ClusterNode locNode = ignite.cluster().localNode();
+
+            log.info(">>> Run test for node [node=" + locNode.id() + ", client=" + locNode.isClient() + ']');
+
+            for (CacheConfiguration ccfg : cacheConfigurations()) {
+                if (locNode.isClient() && ccfg.getCacheMode() == LOCAL)
+                    continue;
+
+                log.info(">>>> Run test for cache " + ccfg.getName());
+
+                c.apply(ignite, ignite.cache(ccfg.getName()));
+            }
+        }
+    }
+
+    /**
+     * Closure with 2 parameters that can throw any exception.
+     *
+     * @param <E1> Type of first closure parameter.
+     * @param <E2> Type of second closure parameter.
+     */
+    public abstract static class CI2Exc<E1, E2> implements CI2<E1, E2> {
+        /**
+         * Closure body.
+         *
+         * @param e1 First closure argument.
+         * @param e2 Second closure argument.
+         * @throws Exception If failed.
+         */
+        public abstract void applyx(E1 e1, E2 e2) throws Exception;
+
+        /** {@inheritDoc} */
+        @Override public void apply(E1 e1, E2 e2) {
+            try {
+                applyx(e1, e2);
+            }
+            catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Closure that can throw any exception.
+     *
+     * @param <T> Type of closure parameter.
+     */
+    public abstract static class CI1Exc<T> implements CI1<T> {
```

Comment:
I see that all inheritors of this closure don't throw checked exceptions.
It can be removed and lambdas can be used instead.

Reply:
This implementation was originally in class `IgniteOptimisticTxSuspendResumeTest`. Will fix it.